app/components/ReproChecklists/ReproChecklists.js

```diff
@@ -82,41 +79,51 @@ const dummyChecklistItems = [
     attachedImages: [],
     attachedURLs: [],
     subChecklists: [],
-    updated: '2024-06-02',
   },
+  {
+    id: '3',
+    statement: 'List of all the languages used and the dependencies of the project',
+    type: 'List',
+    answer: true,
+    scanResult: {},
+    userNotes: [],
+    attachedImages: [],
+    attachedURLs: [],
+    subChecklists:[],
+  }
 ];
 
-function ReproChecklists(props) {
-  const { project, reproChecklist, onUpdatedNote, onDeletedNote, onAddedNote } = props;
-  const [checklistItems, setChecklistItems] = useState(dummyChecklistItems);
+const languages = new Set();
+const dependencies = new Set();
 
-  useEffect(() => {
-    if (project && Array.isArray(project.assets)) {
-      const imageAssets = [];
-
-      const findImageAssets = (asset) => {
-        if (asset.type === 'image') {
-          imageAssets.push(asset);
-        }
-        if (asset.children) {
-          asset.children.forEach(findImageAssets);
+function findAssetLanguagesAndDependencies(asset) {
+  if (asset.contentTypes.includes('code') && asset.type === 'file') {
+    const lastSep = asset.uri.lastIndexOf(path.sep);
```

Comment:
Nothing wrong with this code, but we do have a util method (`AssetUtil.getExtensionFromUri`) that might work here?  Could you see and let me know if it doesn't do what you need?  Sorry too - I know there are a TON of utility methods.  No expectation for you to have known this existed.

https://github.com/StatTag/StatWrap/blob/repro-checklists/app/utils/asset.js#L350

Reply:
oh, didn't know about it. Definitely gonna replace with it!